tests/main/snap-quota-journal/task.yaml

```diff
@@ -0,0 +1,96 @@
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+restore: |
+  echo "Stopping the service"
+  snap stop test-snapd-stressd.logger
+
+execute: |
+  echo "Starting service and verifying that log messages are sent to default journal namespace"
+  snap start test-snapd-stressd.logger
+
+  # The logger service prints 5 messages every 5 seconds, so lets wait 5 seconds to ensure
+  # that messages appear inside journalctl
+  sleep 5
+
+  # Now ask for logs and lets see if we can see messages from 'LogDaemon'
+  # and at the same time lets get the highest message number to compare with later.
+  # The format of the log messages are 'LogDaemon: message number: X'
+  journalctl | MATCH "LogDaemon"
+
+  echo "Create a group with the logger snap in it"
+  snap set-quota group-one --journal-size=16MB --journal-rate-limit=100/10ms test-snapd-stressd
+
+  echo "The systemd slice should be active now"
+  sliceName="snap.$(systemd-escape --path group-one).slice"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=active"
+
+  echo "The service should also still be active"
+  snap services test-snapd-stressd.logger | MATCH "test-snapd-stressd.logger\s+disabled\s+active"
+
+  # wait additional 5 seconds to ensure that the service has been outputting logs
+  sleep 5
+
+  # Now ask for the namespace logs and lets see if we can see messages from 'LogDaemon'
+  journalctl --namespace snap-group-one | MATCH "LogDaemon"
+
+  # Change the size and ensure that the journal is restarted
+  echo "Updating size of journal to ensure size changes"
+  snap set-quota group-one --journal-size=64MB
+
+  # Ensure that the new size is printed in the journa logs. We requested 64MB but
+  # this will most likely print out as 61.0M, so lets be a bit flexible
+  journalctl --namespace snap-group-one | MATCH "max 6[0-9\.]*M,"
+
+  # Ensure that the unit file is correct
+  echo "Ensuring the unit file is correct"
+  systemctl show --property=LogNamespace snap.test-snapd-stressd.logger.service | MATCH "LogNamespace=snap-group-one"
+
+  echo "Removing the quota will stop the slice and the service will be restarted"
+  snap remove-quota group-one
+  snap services test-snapd-stressd.logger | MATCH "test-snapd-stressd.logger\s+disabled\s+active"
+
+  echo "And the service is not in a slice anymore"
+  systemctl show --property=ControlGroup snap.test-snapd-stressd.logger.service | NOMATCH "/$sliceName/snap.test-snapd-stressd.logger.service"
+
+  echo "And the slice is not active anymore"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=inactive"
+
+  # Make sure that the service is running still and now outputting logs
+  # into the original journal.
+  echo "And the service is running and is outputting logs into the original journal"
+  INITIAL_NO=$(journalctl | MATCH "LogDaemon" | tail -n 1 | awk '{print $9}')
+  for i in $(seq 1 5); do
```

Comment:
you can use journal-state match-log for example in this case

Reply:
I am unsure on how to do this loop with match-log, feel free to do any revisions to this PR to help me

Reply:
After take another look, I see match-log cannot be used, but I used get-log instead 
I'll made a change to also retrieve correctly the INITIAL_NO because MATCH does not produce any output in case there is a match

Reply:
Thank you for your help @sergiocazzolato !